src/BloomBrowserUI/bookEdit/js/textOverPicture.ts

```diff
@@ -31,16 +40,237 @@ class TextOverPictureManager {
         });
     }
 
+    public getIsCalloutEditingOn(): boolean {
+        return this.isCalloutEditingOn;
+    }
+
+    public turnOnBubbleEditing(): void {
+        if (this.isCalloutEditingOn === true) {
+            return; // Already on. No work needs to be done
+        }
+        this.isCalloutEditingOn = true;
+
+        Array.from(
+            document.getElementsByClassName("bloom-imageContainer")
+        ).forEach(e => e.classList.add("bloom-hideImageButtons"));
+        // todo: select one of them...make sure this doesn't conflict with any other strategy
+        // for selecting one we just added.
+        // todo: do this for the selected element, not just the first.
+        const textOverPictureElems = document.getElementsByClassName(
+            "bloom-textOverPicture"
+        );
+        if (textOverPictureElems.length > 0) {
+            this.activeElement = textOverPictureElems[0] as HTMLElement;
+            const editable = textOverPictureElems[0].getElementsByClassName(
+                "bloom-editable bloom-visibility-code-on"
+            )[0] as HTMLElement;
+            editable.focus();
+            Comical.convertBubbleJsonToCanvas(
+                this.activeElement!.parentElement!
+            );
+            Comical.activateElement(this.activeElement);
+            Array.from(
+                document.getElementsByClassName("bloom-editable")
+            ).forEach(element => {
+                // tempting to use focusin on the bubble elements here,
+                // but that's not in FF45 (starts in 52)
+
+                // Don't use an arrow function as an event handler here. These can never be identified as duplicate event listeners, so we'll end up with tons of duplicates
+                element.addEventListener(
+                    "focus",
+                    TextOverPictureManager.onFocusSetActiveElement
+                );
+            });
+        }
+    }
+
+    // Event Handler to be called when something relevant on the page frame gets focus.  Will set the active textOverPicture element.
+    public static onFocusSetActiveElement(event: Event) {
+        const focusedElement = event.currentTarget as Element; // The current target is the element we attached the event listener to
+        if (focusedElement.classList.contains("bloom-editable")) {
+            // If we focus something on the page that isn't in a bubble, we need to switch
+            // to having no active bubble element. Note: we don't want to use focusout
+            // on the bubble elements, because then we lose the active element while clicking
+            // on controls in the toolbox (and while debugging).
+            // Todo: check what happens when a bubble is removed.
+            initializeTextOverPictureManager();
```

Comment:
This needs explanation, if it is needed at all. Surely we wouldn't have this event handler if the TOP manager hadn't been initialized? Why do it again?